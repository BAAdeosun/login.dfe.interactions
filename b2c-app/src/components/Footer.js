import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
