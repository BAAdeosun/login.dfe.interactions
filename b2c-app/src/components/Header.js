import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;