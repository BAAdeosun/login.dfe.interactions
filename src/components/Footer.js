import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer