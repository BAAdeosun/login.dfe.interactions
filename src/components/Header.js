import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header