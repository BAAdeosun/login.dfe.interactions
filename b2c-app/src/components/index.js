import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}