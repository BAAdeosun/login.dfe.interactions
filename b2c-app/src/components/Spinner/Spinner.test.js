import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.Spinner />);
});

it('renders correctly with default text', () => {
    const tree = renderer
        .create(<components.Spinner showSpinner={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with custom text', () => {
    const tree = renderer
        .create(<components.Spinner text='custom spinner text' showSpinner={true}></components.Spinner>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('is shown while B2C shows its spinner', () => {
    const tree = renderer.create(<components.Spinner />);

    //spinner not visible
    expect(tree.toJSON()).toMatchSnapshot();

    tree.root.instance.setState({ showB2CSpinner: true });
    //spinner visible because B2C is showing its own spinner
    expect(tree.toJSON()).toMatchSnapshot();

    tree.root.instance.setState({ showB2CSpinner: false });
    //spinner not visible again because B2C has removed its own spinner
    expect(tree.toJSON()).toMatchSnapshot();


    //TODO ver si esto lo puedo hacer usando el mutation observer de verdad y no solo cambiando el estado


});