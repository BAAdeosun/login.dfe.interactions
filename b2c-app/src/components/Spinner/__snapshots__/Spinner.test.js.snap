// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is shown while B2C shows its spinner 1`] = `null`;

exports[`is shown while B2C shows its spinner 2`] = `
<div
  aria-live="assertive"
  className="spinner-container"
  role="alert"
>
  <div
    className="spinner"
  />
  <p
    className="govuk-body"
  >
    Please wait while we process your information.
  </p>
</div>
`;

exports[`is shown while B2C shows its spinner 3`] = `null`;

exports[`renders correctly with custom text 1`] = `
<div
  aria-live="assertive"
  className="spinner-container"
  role="alert"
>
  <div
    className="spinner"
  />
  <p
    className="govuk-body"
  >
    custom spinner text
  </p>
</div>
`;

exports[`renders correctly with default text 1`] = `
<div
  aria-live="assertive"
  className="spinner-container"
  role="alert"
>
  <div
    className="spinner"
  />
  <p
    className="govuk-body"
  >
    Please wait while we process your information.
  </p>
</div>
`;
